misc/i18n/generateTranslation.js

```diff
@@ -0,0 +1,270 @@
+            break;
+          default:
+            console.log('error: trace owned but not located')
+            break;
+        }
+      }
+    }
+  }
+
+  for (const set of relicsetConfig) {
+    output.relicsets[set.SetID] = cleanString(locale, textmap[set.SetName.Hash])
+  }
+
+  for (const lightcone of lightconeConfig) {
+    output.lightcones[lightcone.ID] = cleanString(locale, textmap[lightcone.ItemName.Hash])
+  }
+
+  for (const path of pathConfig) {
+    output.paths[path.ID] = cleanString(locale, textmap[path.BaseTypeText.Hash])
+  }
+
+  for (const element of damageConfig) {
+    output.elements[element.ID] = cleanString(locale, textmap[element.DamageTypeName.Hash])
+  }
+
+  writeFile(`../../public/locales/${locale}/gameData.json`, JSON.stringify(output), (err) => {
+    if (err)
+      console.log(err);
+    else {
+      console.log("File written successfully\n");
+    }
+  })
+}
+
+function cleanString (locale, string) {
+  if (locale !== 'jp') {
+    return string
+  }
+  const regex = /({[^}]*})/g
+  return string.replace(regex, '')
+}
+
+function tbIdToNativeName (id, textmap, pathmap, locale) {
+  const isCaelus = id % 2
+  const path = ((id)=>{
+    const pathIndex = Math.ceil((id - 8000) / 2) - 1
+    const pathT = trailblazerpaths[pathIndex]
+    let hash = ''
+    for (const path of pathmap) {
+      if (path.ID == pathT) {
+        hash = path.BaseTypeText.Hash
+      }
+    }
+    return textmap[hash]
+  })(id)
+  let nativeName = getTbName(locale, isCaelus)
+  return nativeName + ' ' + '(' + path + ')'
+}
+
+function getTbName (locale, isCaelus) {
+  const TB_NAMES = {
+    de: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    en: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    es: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    fr: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    id: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    jp: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    kr: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    pt: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    ru: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    zh: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    th: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+    vi: {
+      stelle: 'Stelle',
+      caelus: 'Caelus'
+    },
+  }
+  if (isCaelus) return TB_NAMES[locale].caelus
+  return TB_NAMES[locale].stelle
+}
+// shoutout to stargazer for these
+function getHash(key) {
+  var hash1 = 5381;
+  var hash2 = 5381;
+  for (let i = 0; i < key.length; i += 2)
+  {
+    hash1 = Math.imul((hash1 << 5) + hash1, 1) ^ key.charCodeAt(i);
+    if (i === key.length - 1)
+      break;
+    hash2 = Math.imul((hash2 << 5) + hash2, 1) ^ key.charCodeAt(i + 1);
+  }
+  return Math.imul(hash1 + Math.imul(hash2, 1566083941), 1);
+}
```

Comment:
Im not following what this is doing here, need some documentation

Reply:
I'll fully admit I have no clue on the what/why for this.
This is what the stargazer guys use to find the hash when it isn't given in the files from Dim's repo